File: data/imported/knowledge/pdf/cpp/merged/snippets/snippet_004.cpp
TEST(ActionsSmoke, ClonesThroughLinkAnnotationAction) {
    Document doc;
    Page page = doc.Pages().Add();
    LinkAnnotation link{page, Rectangle{0.0, 0.0, 100.0, 20.0, false}};

    // Each action type survives the clone done by LinkAnnotation::Action.
    link.Action(NamedAction{PredefinedAction::LastPage});
    ASSERT_NE(link.Action(), nullptr);
    EXPECT_EQ(link.Action()->ToString(), "LastPage");
    EXPECT_NE(dynamic_cast<NamedAction*>(link.Action()), nullptr);

    link.Action(JavascriptAction{"this.print();"});
    EXPECT_EQ(link.Action()->GetECMAScriptString(), "this.print();");
    EXPECT_NE(dynamic_cast<JavascriptAction*>(link.Action()), nullptr);

    SubmitFormAction sf;
    sf.Url(FileSpecification{"https://example.org/x"});
    link.Action(sf);
    auto* got = dynamic_cast<SubmitFormAction*>(link.Action());
    ASSERT_NE(got, nullptr);
    EXPECT_EQ(got->Url().Name(), "https://example.org/x");
}

File: data/imported/knowledge/pdf/java/merged/snippets/snippet_004.java
@Test
    public void border_clear_removesEntry() throws Exception {
        try (Document doc = new Document()) {
            Page page = doc.getPages().add();
            WidgetAnnotation w = new WidgetAnnotation(page, new Rectangle(0, 0, 100, 50));
            w.getCharacteristics().setBorder(Color.fromRgb(0, 1, 0));
            assertNotNull(w.getCharacteristics().getBorder());
            w.getCharacteristics().setBorder(null);
            assertNull(w.getCharacteristics().getBorder());
        }
    }